src/common/src/FileUpload/FileList/FileList.tsx

```diff
@@ -49,24 +49,22 @@ const FileList: React.FC<Props> = ({
 	fileUploadErrorLabel,
 	fileUploadInProgressLabel,
 	...otherProps
-}: Props) => {
-	return (
-		<FileListStyle data-testid="file-list" {...otherProps}>
-			{files.map((file) => (
-				<FileListItem
-					key={file.tmpId}
-					onCancel={() => onCancel(file)}
-					onFileRemove={() => onFileRemove(file)}
-					cancelLabel={cancelLabel}
-					removeLabel={removeLabel}
-					file={file}
-					fileUploadErrorLabel={fileUploadErrorLabel}
-					fileUploadInProgressLabel={fileUploadInProgressLabel}
-				/>
-			))}
-		</FileListStyle>
-	);
-};
+}: Props) => (
+	<FileListStyle data-testid="file-list" {...otherProps}>
+		{files.map((file, index) => (
```

Comment:
Consciously using an index here even though it is considered bad practice. The order of the file list is mostly unchanged between re-renders. There is no unique identifier present in the file object to rely upon